specs/interop/derivation.md

```diff
@@ -87,17 +91,185 @@ The interop network upgrade timestamp defines the timestamp at which all functio
 the consensus rules for an OP Stack based network. On the interop network upgrade block, a set of deposit transaction
 based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
 
-- L1 Attributes Transaction calling `setL1BlockValuesIsthmus`
-- User deposits from L1
-- Network Upgrade Transactions
-  - CrossL2Inbox deployment
-  - L2ToL2CrossDomainMessenger deployment
-  - OptimismMintableERC20Factory deployment
-  - Update CrossL2Inbox Proxy ERC-1967 Implementation Slot
-  - Update L2ToL2CrossDomainMessenger Proxy ERC-1967 Implementation Slot
-  - Update the OptimismMintableERC20Factory Proxy ERC-1967 Implementation Slot
-
-The execution payload MUST set `noTxPool` to `true` for this block.
+The execution payload MUST set `noTxPool` to `true` for this block, excluding user transactions from the block.
+
+This upgrade requires `3_320_000` in gas. In order to upgrade successfully the following invariant MUST be maintained:
+
+```plaintext
+MaxResourceLimit + SystemTxMaxGas + UpgradeGasUsage ≤ L2GasLimit
+```
+
+With MaxResourceLimit set to `20_000_000` and SystemTxMaxGas set to `1_000_000`,
+this mean every upgrading chain MUST have their L2GasLimit set to greater than
+`24_320_000`.
+
+The upgrade transaction details below are based on a nightly release at commit
+hash `71c460ec7c7c05791ddd841b97bcb664a1f0c753`, and will be updated once a
+contracts release is made.
+
+<!-- To regenerate the deploy/upgrade tx docs below, run `./scripts/upgrades/gen_interop_upgrade_tx_specs.sh | pbcopy` and then paste here -->
+
+### CrossL2Inbox Deployment
+<!-- Generated with: ./scripts/run_gen_predeploy_docs.sh --optimism-repo-path ../optimism --fork-name Interop --contract-name CrossL2Inbox --from-address 0x4220000000000000000000000000000000000000 --from-address-nonce 0 --git-commit-hash 71c460ec7c7c05791ddd841b97bcb664a1f0c753 --eth-rpc-url https://optimism.rpc.subquery.network/public --proxy-address 0x4200000000000000000000000000000000000022 --copy-contract-bytecode true -->
+
+The `CrossL2Inbox` contract is deployed.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4220000000000000000000000000000000000000`
+- `to`: `null`
+- `mint`: `0`
+- `value`: `0`
+- `nonce`: `0`
+- `gasLimit`: `420000`
+- `data`: `0x0x6080604052348015600e575f80fd5b...` ([full bytecode](../static/bytecode/interop-cross-l2-inbox-deployment.txt))
+- `sourceHash`: `0x6e5e214f73143df8fe6f6054a3ed7eb472d373376458a9c8aecdf23475beb616`,  
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: CrossL2Inbox Deployment"`
+
+This results in the Interop CrossL2Inbox contract being deployed to
+`0x691300f512e48B463C2617b34Eef1A9f82EE7dBf`, to verify:
+
+```bash
+cast compute-address --nonce=0 0x4220000000000000000000000000000000000000
```

Comment:
Here, how do we ensure that the nonce has not being used in the past already? 
I assume before upgrading the contract we perform a check to know that no collision are possible?

Reply:
You can grab the nonce from an isthmus supported network to know because the address has no known private key and is impersonated only during an upgrade

Reply:
Interop uses deployment addresses `0x422000000000000000000000000000000000000N` with `N` being `0` or `1` for the two deployments. Isthmus used `0x421000000000000000000000000000000000000N`